Google Scholar.js

```diff
@@ -661,6 +579,646 @@ ItemFactory.prototype.saveItemCommonVars = function () {
 	}
 };
 
+
+/*
+ * #########################
+ * ### Utility Functions ###
+ * #########################
+ */
+
+// Returns a promise that resolves (to undefined) after the minimum time delay
+// specified in milliseconds
+function delay(ms) {
+	return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+// Reusable external-search utility for both DOI and ArXiv
+
+/**
+ * Search using external ID
+ *
+ * @constructor
+ * @param {string} searchKey - The property to be used for the ID in the
+ * item-stub object used as the basis of search (e.g. "DOI", "arXiv", etc.)
+ * @param {string} translatorUUID - The id of the search translator
+ * @param {boolean} [doAttach=false] - Whether we should use a workaround to
+ * save attachments
+ * @property {Z.Translate<Z.SearchTranslator>} translator
+ * @property {string} key
+ */
+function ExternalSearch(searchKey, translatorUUID, doAttach = false) {
```

Comment:
So echoing https://github.com/zotero/translators/pull/3062#issuecomment-1621544494, as a general rule, let's just say no constructors in translators. There are likely a few bad examples of this in existing translators (RIS comes to mind), but most translators just use functions and it is good and fine.

There's just too much engineering here. Translators need to be simple and easy to understand. There can't be constructors, methods, pipelines, factory functions, generators — it's just not appropriate for a translator. I understand that you're trying to do some things here that other translators don't have to deal with, but if the logic needs to be slightly less optimized to keep the translator simpler, so be it.